### DISCUSSION TITLE
Debian/RPM Package

### DISCUSSION THREAD

**Discussion by reishoku** [lima-vm/lima#623] [Ideas]
Hi,

I'm considering building official Debian/RPM Package for Lima.

Homebrew/Linuxbrew installs Lima with Permission `reishoku:reishoku` (username+groupname who installed brew with superuser priv), so Lima can't be used in multi-user environment.

Lima can be installed by manually building, but in terms of package management, (I assume) this is not nice.

Any thoughts?

[P.S.] I've successfully built debian package of Lima 0.8.1. ;)

```console
reishoku@Amber ~/Desktop> ls lima*.deb
lima_0.8.1-1_amd64.deb
reishoku@Amber ~/Desktop> LANG=C apt list --installed | grep lima

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

lima/now 0.8.1-1 amd64 [installed,local]
```

**Comment by afbjorklund** [comment]
@reishoku : normally the linuxbrew installation uses `~linuxbrew/.linuxbrew`, so that it can be shared between multiple users

the macbrew version uses either `/usr/local` or `/opt/homebrew`, unless you install it for your user account only (then it can't be shared)

https://docs.brew.sh/FAQ#why-should-i-install-homebrew-in-the-default-location

Having .deb and .rpm would be nice either way, did you add the packaging scripts ?

**Comment by reishoku** [comment]
@afbjorklund :

> normally the linuxbrew installation uses ~linuxbrew/.linuxbrew, so that it can be shared between multiple users

Yes, Linuxbrew installation uses `/home/linuxbrew/.linuxbrew`, but the owner of that directory is the user who installed Linuxbrew with superuser priv.

Fresh installation of linuxbrew:

```console
reishoku@lima-default:/home/reishoku$ ls -la /home/linuxbrew/.linuxbrew/
total 56
drwxr-xr-x 14 reishoku reishoku 4096 Feb  2 15:04 .
drwxr-xr-x  3 root     root     4096 Feb  2 15:04 ..
drwxrwxr-x  2 reishoku reishoku 4096 Feb  2 15:04 Caskroom
drwxrwxr-x  2 reishoku reishoku 4096 Feb  2 15:04 Cellar
drwxrwxr-x  2 reishoku reishoku 4096 Feb  2 15:04 Frameworks
drwxr-xr-x 11 reishoku reishoku 4096 Feb  2 15:05 Homebrew
drwxrwxr-x  2 reishoku reishoku 4096 Feb  2 15:05 bin
drwxrwxr-x  3 reishoku reishoku 4096 Feb  2 15:08 etc
drwxrwxr-x  2 reishoku reishoku 4096 Feb  2 15:04 include
drwxrwxr-x  2 reishoku reishoku 4096 Feb  2 15:04 lib
drwxrwxr-x  2 reishoku reishoku 4096 Feb  2 15:04 opt
drwxrwxr-x  2 reishoku reishoku 4096 Feb  2 15:04 sbin
drwxrwxr-x  6 reishoku reishoku 4096 Feb  2 15:08 share
drwxrwxr-x  3 reishoku reishoku 4096 Feb  2 15:04 var
```

> did you add the packaging scripts ?

I added `debian/control` `debian/copyright` `debian/rules` `debian/changelog`  (and some other files) to the top directory of the repository. Then I ran `debuild -us -uc -b` to generate `.deb` package.

Will create PR.

**Reply by afbjorklund** [reply]
> Yes, Linuxbrew installation uses `/home/linuxbrew/.linuxbrew`, but the owner of that directory is the user who installed Linuxbrew with superuser priv.

Hmm, you are right. Seems like a bug in brew. Obviously /home/linuxbrew should belong to user "linuxbrew", no ?

> The prefix /home/linuxbrew/.linuxbrew was chosen so that users without admin access can ask an admin to create a linuxbrew role account and still benefit from precompiled binaries. If you do not yourself have admin privileges, consider asking your admin staff to create a linuxbrew role account for you with home directory set to /home/linuxbrew.

Seems to be _accessible_ (readable) to everyone, though ? So should be the same situation as those Mac dirs.

**Reply by afbjorklund** [reply]
> Will create PR.

Not sure yet if the project will do packages, or if that is left as an exercise to the distributions. But I guess a PR is one way to find out ?
